common/flatpak-dir.c

```diff
@@ -1970,18 +1970,42 @@ _flatpak_dir_ensure_repo (FlatpakDir   *self,
           GKeyFile *orig_config = NULL;
           g_autoptr(GKeyFile) new_config = NULL;
           g_autofree char *orig_min_free_space_percent = NULL;
+          g_autofree char *orig_min_free_space_size = NULL;
           g_autofree char *orig_add_remotes_config_dir = NULL;
+          const char *min_free_space_size = "500MB";
+          guint64 min_free_space_percent_int;
 
           orig_config = ostree_repo_get_config (repo);
           orig_min_free_space_percent = g_key_file_get_value (orig_config, "core", "min-free-space-percent", NULL);
+          orig_min_free_space_size = g_key_file_get_value (orig_config, "core", "min-free-space-size", NULL);
           orig_add_remotes_config_dir = g_key_file_get_value (orig_config, "core", "add-remotes-config-dir", NULL);
 
-          if (orig_min_free_space_percent == NULL ||
+          if (orig_min_free_space_size == NULL ||
+              orig_min_free_space_percent != NULL ||
               orig_add_remotes_config_dir == NULL)
             new_config = ostree_repo_copy_config (repo);
 
-          if (orig_min_free_space_percent == NULL)
-            g_key_file_set_string (new_config, "core", "min-free-space-percent", "0");
+          /* Scrap previously written min-free-space-percent=0 and replace it with min-free-space-size */
+          if (orig_min_free_space_percent != NULL &&
+              g_ascii_string_to_unsigned (orig_min_free_space_percent, 10,
```

Comment:
If this is going to be upstreamed, note that you’ll need to bump the `GLIB_REQS` in `configure.ac` to `2.54.0`, since that’s when those APIs were added to GLib.

That’s not necessary on Endless, since we’ve backported the APIs to our current version of GLib.